plugins/outputs/groundwork/groundwork.go

```diff
@@ -0,0 +1,363 @@
+package groundwork
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	"strconv"
+
+	"github.com/gwos/tcg/sdk/clients"
+	"github.com/gwos/tcg/sdk/transit"
+	"github.com/hashicorp/go-uuid"
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/internal"
+	"github.com/influxdata/telegraf/plugins/outputs"
+)
+
+const (
+	defaultMonitoringRoute = "/api/monitoring?dynamic=true"
+)
+
+// Login and logout routes from Groundwork API
+const (
+	loginURL  = "/api/auth/login"
+	logoutURL = "/api/auth/logout"
+)
+
+var (
+	sampleConfig = `
+  ## URL of your groundwork instance.
+  url = "https://groundwork.example.com"
+
+  ## Agent uuid for GroundWork API Server.
+  agent_id = ""
+
+  ## Username and password to access GroundWork API.
+  username = ""
+  password = ""
+
+  ## Default display name for the host with services(metrics).
+  # default_host = "telegraf"
+
+  ## Default service state.
+  # default_service_state = "SERVICE_OK"
+
+  ## The name of the tag that contains the hostname.
+  # resource_tag = "host"
+`
+)
+
+type Groundwork struct {
+	Server              string `toml:"url"`
+	AgentID             string `toml:"agent_id"`
+	Username            string `toml:"username"`
+	Password            string `toml:"password"`
+	DefaultHost         string `toml:"default_host"`
+	DefaultServiceState string `toml:"default_service_state"`
+	ResourceTag         string `toml:"resource_tag"`
+	authToken           string
+
+	Log telegraf.Logger `toml:"-"`
```

Comment:
Usually we maintain the order below. But not mandatory...
```suggestion
	Log telegraf.Logger `toml:"-"`

	authToken           string
```

Reply:
Done